### PR TITLE
Refactor: Pass arbitrary data to line-chart-data-loader

### PR DESCRIPTION
**Pull request by stephanwlee**
The loader, previously, was very geared towards runs and tag
combination which can be generalized a little bit. 

Also, while working on this, I factored out the loading logic out of 
the line-chart-data-loader as a reusable behavior and, now, the
line-chart-data-loader is responsible for redrawing the chart.

### CODE REVIEW

File: tensorboard/plugins/scalar/tf_scalar_dashboard/tf-scalar-dashboard.html [hunk 1]
```diff
@@ -138,7 +138,7 @@ <h3>No scalar data was found.</h3>
                           ignore-y-outliers="[[_ignoreYOutliers]]"
                           show-download-links="[[_showDownloadLinks]]"
                           request-manager="[[_requestManager]]"
-                          runs="[[item.runs]]"
+                          load="[[item.items]]"
```
[the remark this answers is not in the document]
Comment (stephanwlee) [reply tensorboard/plugins/scalar/tf_scalar_dashboard/tf-scalar-dashboard.html]: FYI, this property will directly come from the categorization logic and I am not sure how I feel about naming that property "series".

[the remark this answers is not in the document]
Comment (nfelt) [reply tensorboard/plugins/scalar/tf_scalar_dashboard/tf-scalar-dashboard.html]: It just seemed like something more descriptive than "items" would be better.  I think it is generally true that the notion of a "run" or an "experiment + run" combined with a specific tag that it's associated with is a "series" in the sense that it's a single time series of data that can be plotted or displayed.  There's a little precedent for that naming in https://github.com/tensorflow/tensorflow/blob/v1.10.0-rc1/tensorflow/contrib/tensorboard/db/schema.cc#L79

File: tensorboard/components/tf_line_chart_data_loader/data-loader-behavior.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+     * identifiable string. Used for caching purposes.
+     */
+    getDataLoadId: {
+      type: Function,
+      value: () => (datum) => String(datum),
+    },
+
+    /**
+     * A function that takes as inputs:
+     * 1. Implementing component of data-loader-behavior.
```
[the remark this answers is not in the document]
Comment (stephanwlee) [reply tensorboard/components/tf_line_chart_data_loader/data-loader-behavior.ts]: I disagree. The method is implemented and passed by higher order components (i.e., tf-scalar-card). Relying on `this` to be the `tf-line-chart-data-loader` in a method in `tf-scalar-card` seem wrong to me.

[the remark this answers is not in the document]
Comment (nfelt) [reply tensorboard/components/tf_line_chart_data_loader/data-loader-behavior.ts]: Ok, right, fair enough.  I guess the methods would need to access that higher level component and so `this` shouldn't be changed.

File: tensorboard/components/tf_line_chart_data_loader/data-loader-behavior.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+ */
+export const DataLoaderBehavior = {
+  properties: {
+    /**
+     * An unique identifiable string. When changes, it expunges the data
+     * cache.
+     */
+    loadKey: String,
+
+    // List of data to be loaded. A datum is passed to `dataUrl` to get URL
```
[the remark this answers is not in the document]
Comment (stephanwlee) [reply tensorboard/components/tf_line_chart_data_loader/data-loader-behavior.ts]: Done!

**Pull request merged**